src/wp-includes/interactivity-api/interactivity-api.php

```diff
@@ -0,0 +1,134 @@
+<?php
+/**
+ * Interactivity API: Functions and hooks
+ *
+ * @package WordPress
+ * @subpackage Interactivity API
+ */
+
+/**
+ * Processes the directives on the rendered HTML of the interactive blocks.
+ *
+ * This processes only one root interactive block at a time because the
+ * rendered HTML of that block contains the rendered HTML of all its inner
+ * blocks, including any interactive block. It does so by ignoring all the
+ * interactive inner blocks until the root interactive block is processed.
+ *
+ * @since 6.5.0
+ *
+ * @param array $parsed_block The parsed block.
+ * @return array The same parsed block.
+ */
+function wp_interactivity_process_directives_of_interactive_blocks( $parsed_block ) {
+	static $root_interactive_block = null;
+
+	/*
+		* Checks whether a root interactive block is already annotated for
+		* processing, and if it is, it ignores the subsequent ones.
+		*/
+	if ( null === $root_interactive_block ) {
```

Comment:
Spacing looks off here:

```suggestion
	/*
	 * Checks whether a root interactive block is already annotated for
	 * processing, and if it is, it ignores the subsequent ones.
	 */
	if ( null === $root_interactive_block ) {
```